payjoin-io/src/ohttp_keys.rs

```diff
@@ -0,0 +1,89 @@
+use ohttp::KeyConfig;
+use tokio::task::spawn_blocking;
+use url::Url;
+
+use crate::error::{Error, InternalError};
+
+/// Returns the ohttp keys from a payjoin server.
+///
+/// `proxy_endpoint` - The proxy the user want to use when
+/// requesting the ohttp keys from a payjoin server.
+/// `pj_endpoint` - The payjoin server endpoint to fetch the ohttp
+/// keys from.
+pub async fn fetch(proxy_endpoint: Url, pj_endpoint: &Url) -> Result<KeyConfig, Error> {
+    let ohttp_keys_url = pj_endpoint.join("/ohttp-keys")?;
+    let res = spawn_blocking(move || {
+        let proxy = match ProxyServer::new(&proxy_endpoint) {
+            Ok(p) => p,
+            Err(e) => return Err(e),
+        };
+        proxy.get(ohttp_keys_url.as_str()).call().map_err(Error::from)
+    })
+    .await?
```

Comment:
I'm not sure we want to or need to depend on tokio or async operations in this crate yet. I'd prefer we didn't and focused on simple but correct I/O operations and the consumer upstream can still make their own decision on whether or not to use tokio/async-std etc. Having tokio as a feature that exposes async functions may make sense in later versions.